src/App.js

```diff
@@ -45,67 +44,36 @@ const TaskContainer = styled.div`
   }
 `
 
-class App extends React.Component {
-    state = {
-        tasks: [],
-    }
+const App = () => {
+    const { state, dispatch } = useContext(TaskContext);
 
-    componentDidMount() {
+    useEffect(() => {
         const tasks = JSON.parse(localStorage.getItem("tasks"))
-        tasks ? this.setState({tasks: tasks}) : this.setState({tasks: []})
-    }
+        tasks && dispatch(setTask(tasks))
+    }, [])
 
-    componentDidUpdate(prevProps, prevState) {
-            if(prevState.tasks !== this.state.tasks) {
-                localStorage.setItem("tasks", JSON.stringify(this.state.tasks))
-            }
-    }
+    useEffect(() => {
+        localStorage.setItem("tasks", JSON.stringify(state.tasks))
```

Comment:
Actually, it is a bad idea to put smth on useEffect if you can do it on some handler. Will be better to move it in `AddTask` and do it on adding each new task.